controllers/__init__.py

```diff
@@ -273,38 +273,28 @@ def bulk_delete_link_metadata_key(
         return self.db.links.update_many({"_id": {"$in": link_ids}},
                                          update_expr)
 
-    def bulk_upsert_interface_details(
-        self, ids_details: List[Tuple[str, dict]]
+    def upsert_interface_details(
+        self,
+        id_: str,
+        available_tags: dict[list[list[int]]],
+        tag_ranges: dict[list[list[int]]],
     ) -> Optional[dict]:
         """Update or insert interfaces details."""
         utc_now = datetime.utcnow()
-        ops = []
-        for _id, detail_dict in ids_details:
-            ops.append(
-                UpdateOne(
-                    {"_id": _id},
-                    {
-                        "$set": InterfaceDetailDoc(
-                            **{
-                                **detail_dict,
-                                **{
-                                    "updated_at": utc_now,
-                                    "_id": _id,
-                                },
-                            }
-                        ).dict(exclude={"inserted_at"}),
-                        "$setOnInsert": {"inserted_at": utc_now},
-                    },
-                    upsert=True,
-                ),
-            )
-
-        with self.interface_details_lock:
-            with self.db_client.start_session() as session:
-                with session.start_transaction():
-                    return self.db.interface_details.bulk_write(
-                        ops, ordered=False, session=session
-                    )
+        updated = self.db.interface_details.find_one_and_update(
+            {"_id": id_},
+            {
+                "$set": {
+                    "available_vlans": available_tags,
```

Comment:
InterfaceDetailDoc should be refactored accordingly and documented, before performing insertions of major docs, check out https://kytos-ng.github.io/napps/mongodb.html#napps-modules for more details in case you need a review of how that's structured.

Reply:
Refactored and being used.